ldap_auth_provider.py

```diff
@@ -88,6 +93,20 @@ def check_password(self, user_id, password):
             defer.returnValue(False)
         # user_id is of the form @foo:bar.com
         localpart = user_id.split(":", 1)[0][1:]
+        uid_value = localpart
+        default_givenName = localpart
+        if self.ldap_ad_forest:
+            login = localpart
+            domain = self.ldap_default_domain
+
+            if self.ldap_separator not in localpart:
+                if not self.ldap_default_domain:
+                    defer.returnValue(False)
```

Comment:
it would probably be a good idea to log something here, otherwise people are going to be confused when their ldap logins fail silently.